src/Storages/MergeTree/MergeTreeIndexBloomFilter.cpp

```diff
@@ -366,7 +366,31 @@ bool MergeTreeIndexConditionBloomFilter::extractAtomFromTree(const RPNBuilderTre
         }
     }
 
-    return traverseFunction(node, out, nullptr /*parent*/);
+    if (node.isFunction())
+    {
+        /// Similar to the logic of KeyCondition, restrict the usage of bloom filter, in case of func like cast(c=1 or c=9999 as Bool).
+        const std::unordered_set<String> atom_map
```

Comment:
Maybe I miss something but I don't see how the test relates to the functions in l. 374-384.

Reply:
@rschu1ze It's my misunderstanding. This [pr](https://github.com/ClickHouse/ClickHouse/pull/67423) should fix the cast problem as well. So I just add test here.